internal/tiltfile/k8s.go

```diff
@@ -139,16 +143,56 @@ func (s *tiltfileState) filterYaml(thread *starlark.Thread, fn *starlark.Builtin
 		return nil, err
 	}
 
-	match, rest, err := k8s.FilterByMetadataLabels(entities, metaLabels)
-	if err != nil {
-		return nil, err
+	var match, allRest []k8s.K8sEntity
+	match = append(match, entities...)
+	if len(metaLabels) > 0 {
+		match, allRest, err = k8s.FilterByMetadataLabels(match, metaLabels)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if name != "" {
+		var rest []k8s.K8sEntity
+		match, rest, err = k8s.FilterByName(match, name)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(rest) > 0 {
```

Comment:
out of curiosity, can you skip the `if` check and just do a plain append?

Reply:
hmm maybe, let me look.